packages/cactus-plugin-ledger-connector-besu/src/test/typescript/benchmark/run-plugin-ledger-connector-besu-benchmark.ts

```diff
@@ -0,0 +1,239 @@
+  });
+
+  await connector.getOrCreateWebServices();
+  await connector.registerWebServices(expressApp, wsApi);
+
+  return {
+    httpApi: apiClient,
+    apiServer: connector,
+  };
+};
+
+const main = async (opts: { readonly argv: Readonly<Array<string>> }) => {
+  const logLevel: LogLevelDesc = "WARN";
+
+  const gitRootPath = path.join(
+    __dirname,
+    "../../../../../../", // walk back up to the project root
+  );
+
+  console.log("%s gitRootPath=%s", LOG_TAG, gitRootPath);
+
+  const DEFAULT_OUTPUT_FILE_RELATIVE_PATH =
+    ".tmp/benchmark-results/plugin-ledger-connector-besu/run-plugin-ledger-connector-besu-benchmark.ts.log";
+
+  const relativeOutputFilePath =
+    opts.argv[2] === undefined
+      ? DEFAULT_OUTPUT_FILE_RELATIVE_PATH
+      : opts.argv[2];
+
+  console.log(
+    "%s DEFAULT_OUTPUT_FILE_RELATIVE_PATH=%s",
+    LOG_TAG,
+    DEFAULT_OUTPUT_FILE_RELATIVE_PATH,
+  );
+
+  console.log("%s opts.argv[2]=%s", LOG_TAG, opts.argv[2]);
+
+  console.log("%s relativeOutputFilePath=%s", LOG_TAG, relativeOutputFilePath);
+
+  const absoluteOutputFilePath = path.join(gitRootPath, relativeOutputFilePath);
+
+  console.log("%s absoluteOutputFilePath=%s", LOG_TAG, absoluteOutputFilePath);
+
+  const absoluteOutputDirPath = path.dirname(absoluteOutputFilePath);
+  console.log("%s absoluteOutputDirPath=%s", LOG_TAG, absoluteOutputDirPath);
+
+  await fse.mkdirp(absoluteOutputDirPath);
+  console.log("%s mkdir -p OK: %s", LOG_TAG, absoluteOutputDirPath);
+
+  const { apiServer, httpApi } = await createTestInfrastructure({ logLevel });
+
+  const minSamples = 100;
+  const suite = new Benchmark.Suite({});
+
+  const cycles: string[] = [];
+
+  await new Promise((resolve, reject) => {
+    suite
+      .add("plugin-ledger-connector-besu_HTTP_GET_getOpenApiSpecV1", {
+        defer: true,
+        minSamples,
+        fn: async function (deferred: Benchmark.Deferred) {
+          await httpApi.getOpenApiSpecV1();
+          deferred.resolve();
+        },
+      })
+      .on("cycle", (event: { target: unknown }) => {
+        // Output benchmark result by converting benchmark result to string
+        // Example line on stdout:
+        // plugin-ledger-connector-besu_HTTP_GET_getOpenApiSpecV1 x 1,020 ops/sec ±2.25% (177 runs sampled)
+        const cycle = String(event.target);
+        console.log("%s Benchmark.js CYCLE: %s", LOG_TAG, cycle);
+        cycles.push(cycle);
+      })
+      .on("complete", function () {
+        console.log("%s Benchmark.js COMPLETE.", LOG_TAG);
+        resolve(suite);
+      })
+      .on("error", (ex: unknown) => {
+        console.log("%s Benchmark.js ERROR: %o", LOG_TAG, ex);
+        reject(ex);
+      })
+      .run();
+  });
+
+  const data = cycles.join(EOL);
+  console.log("%s Writing results...", LOG_TAG);
+  await fse.writeFile(absoluteOutputFilePath, data, { encoding: "utf-8" });
+  console.log("%s Wrote results to %s", LOG_TAG, absoluteOutputFilePath);
+
+  await apiServer.shutdown();
```

Comment:
Stop the apiServer in case of an error

Reply:
What I've meant is to put this whole login into `try - catch` block, and use `finally` segment to shutdown the apiServer in both successful run and after exception was thrown (this would work since the script is terminating anyway but could cause misleading error logs to be printed).

Reply:
Additionally include here the code to cleanup all resources allocated by `createTestInfrastructure`, most importantly `besuTestLedger` which will remain on the system after the script is done.